packages/autorest.python/autorest/codegen/models/model_type.py

```diff
@@ -229,10 +245,26 @@ def discriminator(self) -> Optional[Property]:
         except StopIteration:
             return None
 
+    @property
+    def discriminator_property(self) -> Optional[Property]:
+        try:
+            return next(
+                p
+                for p in self.properties
+                if p.is_discriminator
+                and isinstance(p.type, ConstantType)
+                and p.type.value == self.discriminator_value
+            )
+        except StopIteration:
+            return None
```

Comment:
```suggestion
for p in self.properties:
    if p.is_discriminator and isinstance(p.type, ConstantType) and p.type.value == self.discriminator_value:
        return p
return None
```
I think it is cleaner compared with list comprehensions

Reply:
lots of `next(...)` are currently used to get first item of some lists, so here is just trying to align with current coding style.